src/rust/ide/enso-protocol/src/file_manager.rs

```diff
@@ -0,0 +1,411 @@
+
+    /// Moves file to another location.
+    #[MethodInput=MoveFileInput,rpc_name="file/move",result=move_file_result,
+    set_result=set_move_file_result]
+    fn move_file(&self, from:Path, to:Path) -> ();
+
+    /// Reads file's content as a String.
+    #[MethodInput=ReadInput,rpc_name="file/read",result=read_result,set_result=set_read_result]
+    fn read(&self, path:Path) -> String;
+
+    /// Gets file's status.
+    #[MethodInput=StatusInput,rpc_name="file/status",result=status_result,set_result=set_status_result]
+    fn status(&self, path:Path) -> Attributes;
+
+    /// Creates a file in the specified path.
+    #[MethodInput=TouchInput,rpc_name="file/touch",result=touch_result,set_result=set_touch_result]
+    fn touch(&self, path:Path) -> ();
+
+    /// Writes String contents to a file in the specified path.
+    #[MethodInput=WriteInput,rpc_name="file/write",result=write_result,set_result=set_write_result]
+    fn write(&self, path:Path, contents:String) -> ();
+
+    /// Watches the specified path.
+    #[MethodInput=CreateWatchInput,rpc_name="file/createWatch",result=create_watch_result,set_result=set_create_watch_result]
+    fn create_watch(&self, path:Path) -> Uuid;
+
+    /// Delete the specified watcher.
+    #[MethodInput=DeleteWatchInput,rpc_name="file/deleteWatch",result=delete_watch_result,
+    set_result=set_delete_watch_result]
+    fn delete_watch(&self, watch_id:Uuid) -> ();
+}
+}
+
+
+
+// =============
+// === Tests ===
+// =============
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use super::FileKind::RegularFile;
+
+    use json_rpc::messages::Message;
+    use json_rpc::messages::RequestMessage;
+    use json_rpc::test_util::transport::mock::MockTransport;
+    use serde_json::json;
+    use serde_json::Value;
+    use std::future::Future;
+    use utils::test::poll_future_output;
+    use utils::test::poll_stream_output;
+    use futures::task::LocalSpawnExt;
+
+    struct Fixture {
+        transport : MockTransport,
+        client    : Client,
+        executor  : futures::executor::LocalPool,
+    }
+
+    fn setup_fm() -> Fixture {
+        let transport = MockTransport::new();
+        let client    = Client::new(transport.clone());
+        let executor  = futures::executor::LocalPool::new();
+        executor.spawner().spawn_local(client.runner()).unwrap();
+        Fixture {transport,client,executor}
+    }
+
+    #[test]
+    fn test_notification() {
+        let mut fixture = setup_fm();
+        let mut events  = Box::pin(fixture.client.events());
+        assert!(poll_stream_output(&mut events).is_none());
+
+        let expected_notification = FilesystemEvent {
+            path : Path::new("./Main.luna"),
+            kind : FilesystemEventKind::Modified,
+        };
+        let notification_text = r#"{
+            "jsonrpc": "2.0",
+            "method": "filesystemEvent",
+            "params": {"path" : "./Main.luna", "kind" : "Modified"}
```

Comment:
hardcoded name of the project again

Reply:
@wdanilo I don't see why it's a problem to have hardcoded values in a test.